quint/src/ToIrListener.ts

```diff
@@ -591,19 +600,66 @@ export class ToIrListener implements QuintListener {
     this.pushApplication(ctx, 'List', args)
   }
 
-  // record constructor, e.g., { name: "igor", year: 2021 }
-  exitRecord(ctx: p.RecordContext) {
-    const names = ctx.IDENTIFIER().map(n => n.text)
-    const elems: QuintEx[] = popMany(this.exprStack, ctx.expr().length)
-
-    const namesAndValues = zipWith(names, elems, (name, elem) => {
+  // field: expr, or ...record
+  exitRecElem(ctx: p.RecElemContext) {
+    const expr = this.exprStack.pop()!
+    if (ctx.IDENTIFIER()) {
+      // field: expr
       const id = this.idGen.nextId()
       this.sourceMap.set(id, this.loc(ctx))
+      const nameEx: QuintEx = {
+        id,
+        kind: 'str',
+        value: ctx.IDENTIFIER()?.text!,
+      }
+      // the 'Tup' id is not needed, as the tuple is unwrapped
+      this.exprStack.push({
```

Comment:
Answers to these questions would be helpful to add to the comment: 

- Why is the tup unwrapped? 
- What is a tup doing here in the firs place?